packages/data-store/src/types/xstate/IAbstractXStateStore.ts

```diff
@@ -0,0 +1,47 @@
+export type SaveStateArgs = {
+  step: string
+  type: string
+  eventName: string
+  state: string
```

Comment:
I am not to happy about the caller having to worry about transforming the state to a string first. Ideally that happens in the plugin itself near the persistence layer. It would be nice if we can figure out whether the state itself from Xstate is already serializable (probably it is). If that is the case then I rather have a mapper internally close to the persistence or datastore layer that does the (de)serialization, so the outside world can just pas in Xstate states. Of course we would have to look at some generic arguments, because we don't want to pull in Xstate at the datastore plugin, so either we would have to recreate some types/interfaces, or pull it in at the Xstate Plugin level. If states however are not really serializable by default, then ideally we do provide some mapper functions in the Xstate Plugin, so that a caller doesn't need to worry about how to (de)serialize to much. Even it is a simple JSON.parse/stringify, I rather see some methods calling JSON.parse/stringify

Reply:
made the state in the entity `string` and in the plugin `unknown`

Reply:
In the current version the store is exposing a string, as that should know nothing about the implementation. The xstate-machine-persistence plugin has a simplified State interface that resembles the State object of xstate but is more generic